lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -352,6 +353,32 @@ Use by both B<create()> and B<update()> method.
 sub _prepare_settings {
     my ($self, $table, $entry) = @_;
     my $validation = $self->validation;
+    my $hp;
+    # accept both traditional application/x-www-form-urlencoded parameters
+    # with hash entries having key names encoded like settings[value1]
+    # (see doc at the end of HashedParams.pm)
+    # as well as modern application/json encoded hashes
+    my $error;
+    if ($self->req->headers->content_type =~ /^application\/json/) {
+        try {
+            $hp = decode_json($self->req->body);
+        }
+        catch {
+            $error = $_;
+        };
+        # of course stupid perl doesn't let you return directly from catch
+        # as that would actually lead to readable code
```

Comment:
Totally unnecessary comment!

Reply:
Well it explains why the code needs an odd workaround around a missing language feature at this locaton.

But at least I didn't add a 20 line rant like the person writing the open source PSD implementation :P
https://github.com/zepouet/Xee-xCode-4.5/blob/master/XeePhotoshopLoader.m#L108

Reply:
It could have explained that without saying "stupid perl"

Reply:
This looks very unprofessional to me.